### PR TITLE
[backend] Add tests for value field encryption in connector configurations (#7328)

### PR DESCRIPTION
**Pull request by maelv-filigran**
### Proposed changes

* Added comprehensive test coverage for password field encryption in connector configurations, including:
  - Tests for single and multiple password field encryption
  - Validation of encrypted vs unencrypted password fields
  - Handling of optional password fields with encryption
  - Edge cases with special characters and default values
  
### Related issues
* Addresses connector configuration security improvements
* #7328

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

The test additions significantly improve the coverage for password field encryption in connector configurations. The new test file `encryption-test.ts` provides dedicated tests for encryption functionality, while existing test files have been enhanced with additional test cases to validate the proper handling of encrypted fields.

### CODE REVIEW

File: opencti-platform/opencti-graphql/tests/01-unit/domain/catalog-domain/encryption-test.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      const encrypted2 = processPasswordConfigurationValue(password, TEST_RSA_PUBLIC_KEY);
+
+      // Different encrypted values due to random AES key/IV
+      expect(encrypted1).not.toBe(encrypted2);
+
+      // Both should still be valid encrypted formats
+      const buffer1 = Buffer.from(encrypted1, 'base64');
+      const buffer2 = Buffer.from(encrypted2, 'base64');
+      expect(buffer1[0]).toBe(0x01);
+      expect(buffer2[0]).toBe(0x01);
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/tests/01-unit/domain/catalog-domain/encryption-test.ts:91]: We might have a <0.000000001% chance of random AES key collide, so this test might have a flaky run in the next billion years, but I guess we can live with that 🤓

**Review by JeremyCloarec — APPROVED** [review]
Awesome work !
